features/api/backend_usages/index.feature

```diff
@@ -46,8 +46,9 @@ Feature: Product > Integration > Backends
 
     Scenario: Editing a backend config
       When they go to the backends of product "My API"
-      And follow "Edit config with Backend 2"
-      Then the current page is the edit usage config page between "My API" and "Backend 2"
+      Then there is a link "Edit config with Backend 2" to the edit usage config page between "My API" and "Backend 2"
+      When they follow "Edit config with Backend 2"
+      And the current page is the edit usage config page between "My API" and "Backend 2"
```

Comment:
Because the proper order is `Given`, (`And`), `When`, (`And`), `Then`, (`And`). `Then` is the assertion and should be at the end.

Reply:
Order is this. But you may have multiple sentences in a scenario. 
```
there is a link "Edit config with Backend 2" to the edit usage config page between "My API" and "Backend 2"
```
This is kind of a check, an assertion. So reads better to be a `Then`.

Reply:
We can use whatever word we want but I'm afraid Joan is right, `Then` goes after `When` not before. If there is a need for multiple checks, then there should be multiple scenarios. Anyway, we don't really use Cucumber properly so who cares.
This step: `Then there is a link "Edit config with Backend 2" to the edit usage config page between "My API" and "Backend 2"` doesn't add anything to the scenario because it is clicked right after, therefore the link should exist.

Reply:
>  If there is a need for multiple checks, then there should be multiple scenarios.

Even in unit tests we have multiple assertions. For me the point is to have subjectively simple enough scenarios. If we try to apply certain rules to the extreme, that often times defeats the purpose of said rules [1].

In this case though, that's checking the same thing, just at different points. I explained this 

[1] I'm yet to see this rule described somewhere.

Reply:
> I don't really understand what you want to achieve with this extra step, because following the "Edit config with Backend 2" should fail if there is no link. No need to assert there is a link.

We spoke already on slack. Here's it again
> ... the point is in debuggability. On failure we collect a screenshot, html and some logs. Once we click, we are on another page and the page where the issue can be understood is already gone. I couldn't understand the issue before I added the step.The issue is that it was not locally reproducible so I had to only use the artifacts to debug.

Basically now the screenshot and the html is from the 404 page and there is no useful information in those.

The step checks not only existence of a link with that title but also the target of the link. It reuses existing methods for that purpose.

Reply:
> IMHO the focus right now should be to use existing steps and reducing the amount of "1 use" steps that make the test suite hard to maintain.

I would argue that this step would be useful at any place we currently click on a link and check the URL. Because if we end up on a different URL, we have already lost the original HTML that led us here and have no information why it happened.